packages/aragon-cli/src/lib/apm/grantNewVersionsPermission.js

```diff
@@ -0,0 +1,76 @@
+const APM = require('@aragon/apm')
+const ACL = require('./util/acl')
+
+module.exports = async (
+  web3,
+  apmRepoName,
+  apmOptions,
+  grantees,
+  progressHandler,
+  { gasPrice }
+) => {
+  if (grantees.length === 0) {
+    throw new Error('No grantee addresses provided')
+  }
+
+  if (progressHandler) {
+    progressHandler(1)
+  }
+
+  // Ensure the ens-registry property is present,
+  // and available with the name "ensRegistryAddress".
+  if (!apmOptions.ensRegistryAddress) {
+    if (apmOptions['ens-registry']) {
+      apmOptions.ensRegistryAddress = apmOptions['ens-registry']
+    } else {
+      throw new Error('ens-registry not found in given apm options.')
+    }
+  }
+
+  const apm = await APM(web3, apmOptions)
+  const acl = ACL(web3)
+
+  if (progressHandler) {
+    progressHandler(2)
+  }
+
+  const repo = await apm.getRepository(apmRepoName)
+  if (repo === null) {
+    throw new Error(
+      `Repository ${apmRepoName} does not exist and it's registry does not exist`
+    )
+  }
+
+  /* eslint-disable-next-line */
+  for (const address of grantees) {
+    if (progressHandler) {
+      progressHandler(3, address)
+    }
+
+    // Decode sender
+    const accounts = await web3.eth.getAccounts()
+    const from = accounts[0]
+
+    // Build transaction
+    const transaction = await acl.grant(repo.options.address, address)
+
+    transaction.from = from
+    transaction.gasPrice = gasPrice
+    // the recommended gasLimit is already calculated by the ACL module
+
+    try {
+      const receipt = await web3.eth.sendTransaction(transaction)
+      if (progressHandler) {
+        progressHandler(4, receipt.transactionHash)
+      }
+    } catch (error) {
+      if (progressHandler) {
+        progressHandler(5, error)
```

Comment:
This is dangerous as the error could easily be ignored / not handled properly. The progressHandler should only deal with informative data updates, not stopping errors. Instead, I would advise using custom error classes https://javascript.info/custom-errors.

As example please look at this implementation
- Definition: https://github.com/aragon/aragon-cli/pull/893/files#diff-86adbef0eb8780f10a545fcba7442b0dR13
- Usage: https://github.com/aragon/aragon-cli/pull/893/files#diff-f8a8bf84d63a26b560eda2f22a89de9dR80

Reply:
I agree. I removed the error handling from the `progressHandler`. It will indeed probably be a good idea to use a custom error once we start focusing on the SDK behavior.

Reply:
Thank you so much